test/refactoring/ConvertAsync/convert_bool.swift

```diff
@@ -461,4 +459,21 @@ func testConvertBool() async throws {
   // OBJC-BOOL-WITH-ERR2-NEXT:   print("neat")
   // OBJC-BOOL-WITH-ERR2-NEXT:   print("neato")
   // OBJC-BOOL-WITH-ERR2-NEXT: }
+
+  // RUN: %refactor-check-compiles -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):3 -I %S/Inputs -I %t -target %target-triple %clang-importer-sdk-nosource | %FileCheck -check-prefix=OBJC-BOOL-WITH-ERR-FALLBACK %s
```

Comment:
Good point regarding setting `success` in the different branches! I can look into doing that, should be reasonably straightforward.

The reason why the fallback is being used is because we have an unhandled condition mixed with a classified condition, and we have an `else` block, so we don't know what to do with it, as the `else` block could be run under success or failure.

Reply:
Added logic to bind the `success` variable, how does it look?

Reply:
LGTM, thanks :)